### DISCUSSION TITLE
[Feature] Rework the (dark mode) design

### DISCUSSION THREAD

**Discussion by orbatschow** [immich-app/immich#1898] [Feature Request]
### The feature

Hi,

thanks for the awesome app. As I am not a designer by myself I can provide limited help on this topic. However there is a fundamental issue at the moment with the design of immich from my point of view. The complete black design is pretty hard on the eye - on my computer and also on my phone. It would be great to introduce/change the current theming to something like:

- Light
- Dark
- Darker

Now the main reason why the design is not a good fit (from my point of view):

immich is a photo/library app. Of course it is used to backup photos, but I think it is also mainly used to scroll through and view photos. And the scrolling is a problem on (non flagship) smartphones. There is a phenomena called "oled smearing", that will cause the black to smear purple when scrolling. This is pretty hard on the eyes. 

I propose three solutions here:

- Change the current dark theme to a gray background
- Introduce a new design called Gray or something similar
- Make the apps/web version customizable, a good example is e.g. [boost]https://play.google.com/store/apps/details?id=com.rubenmayayo.reddit&hl=en&gl=US) as it allows the user to basically change every color that is used within the app

Other sources: https://medium.com/lookup-design/designing-a-dark-theme-for-oled-iphones-e13cdfea7ffe

I would like to emphasize that I really like immich and all the love that went into this!

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
There is a dark/light mode switch on the top right bar of the web app. For the mobile app, you can change the theme in the settings.

**Reply by orbatschow** [reply]
Thanks for the advice! I know that there is a light mode, but my intention was to start a discussion about a dark mode, that wont cause the smearing and also is easier on the eyes than pure black.

**Reply by zackpollard** [reply]
I'm not sure this is something we will be looking into supporting right now. With the way that the web app is laid out, adding a third theme would actually be a substantial amount of work and support effort. Every theme variation we add is extra work when we add new features or change existing ones. Down the road I could see us potentially adding the ability for custom themes, but wouldn't expect that to come anytime soon. I do understand the issue with the oled smearing, but I haven't noticed it on my phone (Pixel 6 Pro, OLED @ 120Hz) and would say it's probably less of an issue on web where oled screens are less commonly found.

**Reply by orbatschow** [reply]
I own a Nothing Phone 1 and it has the same OLED specs as the Pixel 6 Pro and is affected by this problem. It is not only limited to mid range prices smartphones, the article describes, that top of the line iPhones are also affected. I'm sure that the changes require a lot of effort, I just started the conversation to make maintainers aware of the problem and maybe it will also influence the architectural decisions regarding design choice for immich in the future.

**Comment by martyfuhry** [comment]
The default "black" color in Immich's dark mode is indeed "true black" (0, 0, 0). By default, Flutter uses #303030 has a background color for dark themes, which is much less dark. I didn't know about OLED smearing; that is very interesting! I'm not much of a designer, either; @alextran1502 has done most of the design (which looks great).

It's quite a bit of contrast, though, and maybe it could be lowered from true black to something a little less intense.
